### DISCUSSION TITLE
dce driver

### DISCUSSION THREAD

**Discussion by cbldlpl** [NVIDIA/open-gpu-kernel-modules#465] [Q&A] [answered]
I feel confuse about dce in g_dce_client_nvoc.h,
First,what's the meaning of "DCE",it is  an abbreviation？
Second,what function does it serve?
Third,the code about it is open source or not？
I couldn‘t find documents about it.
Eager for a reply, I'm a beginner but would love to improve

**Comment by mtijanic** [comment, marked as answer]
Hey there, welcome! "DCE" stands for "Display Core Engine".. I think. One can never be sure with these acronyms. :)

DCE exists on some Tegra chips to handle the display side of things, but architecturally it's more or less exactly like a GSP (just with less responsibilities, and not RISC-V). You'll probably see references in the code to a "DCE client" just like we have a "GSP client". The two are mutually exclusive.

The DCE client driver is built from the same codebase as the GSP client (this driver), but due to some internal process inefficiencies the DCE code is not part of this repo. We hope to merge them in the future so you could just `make dce` or something.

In the meantime, the DCE code is available for download at https://developer.nvidia.com/embedded/jetson-linux-archive. You'll see a lot of the same files as here.


If you want to do some internet archeology, here's an article from when DCE was open sourced, you can probably find a lot more chatter about it from that time period. https://www.phoronix.com/news/NVIDIA-Kernel-Driver-Source

**Reply by cbldlpl** [reply]
Thank you！You just save myyyyyy day！
